Automattic/jetpack
# Add Likes and Sharing sidebar as a Gutenberg extension

#### Changes proposed in this Pull Request:

This PR converts the Likes and Sharing metabox to a Gutenberg sidebar extension, moving these settings out of the main sidebar and into the Jetpack section. It is mostly a port of https://github.com/Automattic/wp-calypso/pull/29744, with the addition of `__back_compat_meta_box => true` to disable 
the old metabox within Gutenberg.

It also adds some logic so that the checkboxes aren't displayed unless their corresponding modules are active (because the extension will be added if *either* Sharing or Likes is activated, but we don't want to show checkboxes for a disabled setting).

Fixes #10199

#### Testing instructions:

1. Double-check [sharing button prerequisites](https://github.com/Automattic/wp-calypso/pull/29744#issuecomment-459443781).
1. Make sure beta blocks are enabled with `define( 'JETPACK_BETA_BLOCKS', true );` in your `wp-config.php`.
1. Compile the blocks with `yarn build-extensions` in the checked-out branch.
1. Load a new or existing post in the block editor
1. The Likes and Sharing checkboxes should no longer be visible in the post sidebar under Discussion.
1. Under the Jetpack sidebar, there should be a Likes and Shares section with checkboxes to toggle these options:

<img width="299" alt="screen shot 2019-01-24 at 3 44 01 pm" src="https://user-images.githubusercontent.com/52152/51707327-e8355e80-1fee-11e9-9e90-4a915f255728.png">

5. Make sure you can set/unset them and that these settings are appropriately reflected on the post (you can check via the API console or the classic editor, or by just viewing the post on the front-end).

6. Also test by disabling Likes globally and then toggling them ON on a per-post basis.

7. In the Classic Editor, verify that the checkboxes still show up and behave correctly.

8. **New! Disable one or both modules and verify that the checkboxes are shown or hidden as appropriate.**

#### Proposed changelog entry for your changes:

* Convert Likes and Sharing metabox to a Gutenberg plugin
